mlrun/render.py

```diff
@@ -96,20 +99,33 @@ def link_html(text, link=""):
 
 
 def artifacts_html(x, pathcol="path"):
-    if not x:
+    artifact_dicts, output_path = x
+    if not artifact_dicts:
         return ""
     html = ""
-    for i in x:
-        # support legacy format
-        if pathcol in i:
-            link, ref = link_to_ipython(i[pathcol])
-        else:
-            link, ref = link_to_ipython(i["spec"][pathcol])
 
-        if "key" in i:
-            key = i["key"]
+    for artifact_dict in artifact_dicts:
+        # if generate_target_path_from_artifact_hash is True, we need to resolve the target path from the artifact
+        # hash, otherwise we can use the pathcol as is
+        if (
+            pathcol == "target_path"
+            and config.artifacts.generate_target_path_from_artifact_hash
+        ):
+            artifact = mlrun.artifacts.dict_to_artifact(artifact_dict)
+            target_path = artifact.resolve_target_path(artifact_path=output_path)
+            link, ref = link_to_ipython(target_path)
+
+        else:
+            # TODO: remove this in 1.5.0 once we no longer support legacy format
+            if pathcol in artifact_dict:
```

Comment:
you can also use the function `is_legacy_artifact`, since this is what the condition means.